lms/djangoapps/certificates/tests/test_api.py

```diff
@@ -348,6 +348,23 @@ def test_xqueue_submit_task_error(self):
         self.assertEqual(cert.status, 'error')
         self.assertIn(self.ERROR_REASON, cert.error_reason)
 
+    def test_generate_user_certificates_with_unverified_cert_status(self):
+        """
+        Generate user certificate will not raise exception in case of certificate is None.
+        """
+        # generate certificate with unverified status.
+        GeneratedCertificateFactory.create(
+            user=self.student,
+            course_id=self.course.id,
+            status=CertificateStatuses.unverified,
+            mode='verified'
```

Comment:
I can not understand this test. We are creating a certificate with `verified` mode which is available in `_valid_states_`.

Should we be testing a user with `unverified` cert and expecting the following log?
https://github.com/edx/edx-platform/blob/ab0cbfffeb7ac27fe30fab50472d9d39df09194c/lms/djangoapps/certificates/queue.py#L284-L297